src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.cs

```diff
@@ -144,6 +144,15 @@ public void AddDependencyLocation(string originalPath)
         {
             CheckIfDisposed();
 
+            lock (_guard)
+            {
+                if (_originalPathInfoMap.TryGetValue(originalPath, out var originalPathInfo))
+                {
+                    Debug.Assert(GeneratedPathComparer.Equals(_originalPathInfoMap[originalPath].ResolvedPath, originalPathInfo.ResolvedPath));
```

Comment:
I didn't fully understand what this assert is verifying.

Reply:
I tried to match the code in the else clause of the TryAdd call below it, but that's pretty much not useful in this case